test/test-ws.js

```diff
@@ -0,0 +1,90 @@
+
+var assert = require('assert');
+var extend = require('util')._extend;
+var WebSocketChannel = require('../ws-channel');
+var WebSocketServer = require('../ws-server');
+
+var ctlUri = 'ws://127.0.0.1:9999/';
+
+var isParent = process.argv[2] !== 'child';
+
+var gotRequest = 0;
+var gotResponse = 0;
+var gotNotification = 0;
+
+process.on('exit', function() {
+  assert(gotRequest === 1);
+  assert(gotResponse === 1);
+  assert(gotNotification === 1);
+
+  // Print only if we're the parent - not the child.
+  if (isParent)
+    console.log('PASS');
+});
+
+
+if (isParent) {
+  // Parent
+  var server = WebSocketServer.getServer(ctlUri);
+  var channel = server.createChannel(onServerRequest, onServerNotification);
+
+  function onServerRequest(message, callback) {
+    console.log('server got', message);
+    assert(message.cmd === 'serverRequest');
+    gotRequest++;
+    callback({cmd: 'serverResponse'});
+  }
+
+  function onServerNotification(message) {
+    console.log('server got', message);
+    assert(message.cmd === 'serverNotification');
+    gotNotification++;
+  }
+
+  var env = extend({ MESH_TOKEN: channel.getToken() }, process.env_);
+  require('child_process').fork(process.argv[1],
+                                ['child'],
+                                { stdio: 'inherit', env: env });
```

Comment:
Not sure what it is about iojs that makes it not necessary, but under node 0.10 and 0.12 this test never exits. Seems to be something to do with the server. Doing a `server.close()` after the child exits seems to do the trick for me locally.